query-engine/core/src/interpreter/query_interpreters/nested_read.rs

```diff
@@ -24,8 +26,18 @@ pub async fn m2m<'a, 'b>(
                 .projections(&parent_model_id)?
         }
     };
-
+    if parent_ids.is_empty() {
+        return Ok(ManyRecords::new(
+            query.selected_fields.names().map(|n| n.to_string()).collect(),
```

Comment:
Make this a function on `ManyRecords`, something like `pub fn empty(selected_fields: ...)`.

Reply:
Will do.